internal/services/azure_reservation_service.go

```diff
@@ -34,13 +34,16 @@ func CreateAzureReservation(w http.ResponseWriter, r *http.Request) {
 	pkDao := dao.GetPubkeyDao(r.Context())
 	rDao := dao.GetReservationDao(r.Context())
 
-	// Check for preloaded region
-	if payload.Location == "" {
-		payload.Location = "eastus_1"
-	}
-	if !preload.AzureInstanceType.ValidateRegion(payload.Location) {
-		renderError(w, r, payloads.NewInvalidRequestError(r.Context(), "Unsupported location", ErrUnsupportedRegion))
-		return
+	// validate region
+	// it needs to be in region format, but also
+	if payload.Location != "" && !preload.AzureInstanceType.ValidateRegion(payload.Location+"_1") {
+		// Location format is accepted now for backwards compatibility, but we should deprecate it
+		if preload.AzureInstanceType.ValidateRegion(payload.Location) {
+			logger.Warn().Msgf("Azure region passed with location suffix (%s), this is deprecated behaviour format", payload.Location)
```

Comment:
⚠️ change for the future, we now validate region, not location.

In the past, we've validated for location, but then actually ignored the param down the line and forced eastus, but we actually need regions, not locations in all the calls (even tho azure calls them Location 🤦)

Reply:
Does this need to be part of this PR? Also I am not sure if I am following here. Can you explain?

Reply:
Well, we required region+zone in the past, but then we dropped the value and  used hard coded `eastus`, so it worked, but our client expects only region (without the zone suffix).

So this patch accepts both, but when user passes the version with suffix, we now warn about it, so we can keep track of how many usages of that there is.

Reply:
Ok this is fine, if you want to go an extra mile I think you can refactor `ValidateRegion` to accept variadic string and concatenate by `_` because this is exactly how it works for all the clouds.

Reply:
It's actually dash (`-`) for EC2 and GCP and underscore `_` for Azure :face_exhaling: xD